pgrx-sql-entity-graph/src/pgrx_sql.rs

```diff
@@ -403,19 +403,27 @@ impl PgrxSql {
 
     pub fn to_sql(&self) -> eyre::Result<String> {
         let mut full_sql = String::new();
-        for step_id in petgraph::algo::toposort(&self.graph, None).map_err(|e| {
-            eyre!("Failed to toposort SQL entities, node with cycle: {:?}", self.graph[e.node_id()])
-        })? {
-            let step = &self.graph[step_id];
 
-            let sql = step.to_sql(self)?;
-            full_sql.push_str(sql.trim_matches('\n'));
+        for nodes in petgraph::algo::tarjan_scc(&self.graph).iter().rev() {
+            let mut inner_sql = Vec::with_capacity(nodes.len());
```

Comment:
We should probably comment that using this algo is load-bearing for run-to-run ordering.

Reply:
( Especially because as far as we can tell this _isn't_ necessarily the case from executing `tarjan_scc`, it just is almost always so in the current implementation. )

Reply:
done